src/mono/wasm/debugger/BrowserDebugProxy/EvaluateExpression.cs

```diff
@@ -47,7 +47,8 @@ private sealed partial class ExpressionSyntaxReplacer : CSharpSyntaxWalker
             private int visitCount;
             public bool hasMethodCalls;
             public bool hasElementAccesses;
-            internal List<string> variableDefinitions = new List<string>();
+            public bool hasExpressionStatement;
+            internal List<(string idName, JObject obj, string definition)> variableDefinitions = new ();
```

Comment:
make this a record, because where this tuple is being used we don't enforce field names. And it is easy to mix up members then.